api/util.go

```diff
@@ -0,0 +1,93 @@
+package api
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+	"os"
+
+	"github.com/go-resty/resty/v2"
+
+	"github.com/ovalfi/go-sdk/helpers"
+	"github.com/ovalfi/go-sdk/model"
+)
+
+func (c *Call) makeRequest(ctx context.Context, path, method string, params, formData map[string]interface{}, requestBody, responseBody interface{}) error {
+	endpoint := fmt.Sprintf("%s%s", c.baseURL, path)
+
+	log := c.logger.With().Str("method", method).Str("endpoint", endpoint).Logger()
+	log.Info().Msg("starting...")
+
+	var (
+		err    error
+		res    *resty.Response
+		errRes model.ErrorResponse
+	)
+
+	client := c.client.R().
+		SetAuthToken(c.bearerToken).
+		SetHeader(model.RequestIDHeaderKey, helpers.GetRequestID(ctx)).
+		SetResult(&responseBody).
+		SetError(&errRes).
+		SetContext(ctx)
```

Comment:
this won't work.

you need to rename the `ErrorResponse` to `GenericResponse`. if its an error or not, the response matches. It's the Data object in the response you need to map the `responseBody` to.

In other words, test your work sir.

Reply:
The data mapping will happen in the individual function that calls this makeRequest function